mathesar/views/api.py

```diff
@@ -85,9 +89,19 @@ class RecordViewSet(viewsets.ViewSet):
 
     def list(self, request, table_pk=None):
         paginator = TableLimitOffsetPagination()
-        records = paginator.paginate_queryset(self.queryset, request, table_pk)
-        serializer = RecordSerializer(records, many=True)
-        return paginator.get_paginated_response(serializer.data)
+        filter_form = RecordListFilterForm(request.GET)
+        if filter_form.is_valid():
+            try:
+                records = paginator.paginate_queryset(
+                    self.queryset, request, table_pk,
+                    filters=filter_form.cleaned_data['filters']
+                )
+            except (FieldNotFound, BadFilterFormat) as e:
+                raise ValidationError({'filters': e})
+            serializer = RecordSerializer(records, many=True)
+            return paginator.get_paginated_response(serializer.data)
+        else:
+            raise ValidationError(filter_form.errors)
```

Comment:
Wouldn't you want to raise the same error format for all exceptions?

Reply:
I think the error formats do work out the same, `form.errors` should contain a dictionary with fields as keys and errors as values.